nvflare/fuel/flare_api/api_spec.py

```diff
@@ -61,6 +65,24 @@ class AuthorizationError(BaseException):
     pass
 
 
+class NoClientsAvailable(BaseException):
+    pass
+
+
+class ClientsStillRunning(BaseException):
+    pass
+
+
+class InvalidTarget(BaseException):
+    pass
```

Comment:
Instead of making each one an individual class, maybe we can combine them into one class, and add a "class type" field to indicate what class it is. In this case, we don't have to add a new class for each new type / scenario we need to process.

Reply:
Not sure. The API has been designed to use exception for different error conditions.

Reply:
Maybe we could use one exception (e.g. FlareApiError) for all cases. This exception accepts a status_code.
It does get tedious to add a new exception type and a new status code every time we have a new error condition.

Reply:
Just checked the codebase, unfortunately some of the defined exception types have been used in app code and example notebooks. If we change to use one exception type, all those places would have to change.

So for now let's stay with this.